experimenter/experimenter/features/tests/test_nimbus_fml_loader.py

```diff
@@ -319,11 +284,53 @@ def test_return_no_errors_for_invalid_application(
         application = "badapp"
         channel = "release"
         with self.assertLogs(level="ERROR") as log:
-            loader = NimbusFmlLoader(application, channel, self.TEST_BASE_PATH)
+            loader = NimbusFmlLoader(application, channel)
             test_blob = json.dumps({"features": {"new-feature": {"enabled": "false"}}})
 
             result = loader.get_fml_errors(test_blob, "my_feature_id")
 
             self.assertEqual(loader.application, None)
             self.assertEqual(result, [])
             self.assertIn("Nimbus FML Loader: Invalid application", log.output[0])
+
+    @patch(
+        "experimenter.features.manifests.nimbus_fml_loader.NimbusFmlLoader.__init__",
+    )
+    def test_loaders_are_cached(self, mock_loader):
+        mock_loader.return_value = None
+        application = "fenix"
+        channel = "release"
+        loader = NimbusFmlLoader.create_loader(application, channel)
+        mock_loader.assert_called_once()
+
+        cached_loader = NimbusFmlLoader.create_loader(application, channel)
+        mock_loader.assert_called_once()
+        self.assertEqual(loader, cached_loader)
+
+    @patch(
+        "pathlib.Path.exists",
+    )
+    def test_file_paths_are_cached(self, mock_path_exists):
+        mock_path_exists.return_value = True
+        loader = self.create_loader()
+        file_path = loader._get_local_file_path()
+        mock_path_exists.assert_called_once()
+
+        cached_file_path = loader._get_local_file_path()
+        mock_path_exists.assert_called_once()  # not called a second time
+        self.assertEqual(file_path, cached_file_path)
+
+    @patch(
+        "rust_fml.FmlClient.__init__",
+    )
+    def test_clients_are_cached(self, mock_create_client):
+        mock_create_client.return_value = None
+        channel = "release"
+        loader = self.create_loader()
+
+        client = loader._create_client(self.TEST_MANIFEST_PATH, channel)
+        mock_create_client.assert_called_once()
+
+        cached_client = loader._create_client(self.TEST_MANIFEST_PATH, channel)
+        mock_create_client.assert_called_once()  # not called a second time
+        self.assertEqual(client, cached_client)
```

Comment:
@eliserichards and I chatted about this being generally unnecessary since it's really more testing the cache decorator than it is testing our logic, so you don't really need to test this here.